cacheDependencyManagers/cacheDependencyManager.js

```diff
@@ -164,13 +166,20 @@ CacheDependencyManager.prototype.extractDependencies = function (cachePath, call
     callback();
   }
 
-  var extractor = tar.Extract({path: targetPath})
-                     .on('error', onError)
-                     .on('end', onEnd);
+  if (compressedCacheExists) {
+    var extractor = tar.Extract({path: targetPath})
+        .on('error', onError)
+        .on('end', onEnd);
 
-  fs.createReadStream(cachePath)
-    .on('error', onError)
-    .pipe(extractor);
+    fs.createReadStream(cachePath)
+        .on('error', onError)
+        .pipe(extractor);
+  } else {
+    fstream.Reader(cachePath)
+        .on('error', onError)
+        .on('end', onEnd)
+        .pipe(fstream.Writer(targetPath));
```

Comment:
For what is worth, this method of copying a directory structure is really slow. On my machine (with an SSD) :

- `time cp -R ~/.packages-cache ./node_modules` : 13s
- `time npm-cache npm` : 1m 13s

This is a first shot, and good enough for this PR. But I may submit another PR in the future to improve performances of directory copy.